pkg/webhook/nad/mutator.go

```diff
@@ -114,27 +109,15 @@ func (m *Mutator) ensureLabels(nad *cniv1.NetworkAttachmentDefinition, oldConf,
 			}}, nil
 	}
 
-	// Ignore untagged network because we don't need to do more operation if the last network type is untagged network
-	if oldConf.Vlan != 0 && newConf.Vlan == 0 {
-		labels[utils.KeyLastNetworkType] = string(utils.L2VlanNetwork)
-	}
-
-	if newConf.Vlan != 0 {
-		labels[utils.KeyNetworkType] = string(utils.L2VlanNetwork)
-		labels[utils.KeyVlanLabel] = strconv.Itoa(newConf.Vlan)
-	} else {
-		labels[utils.KeyNetworkType] = string(utils.UntaggedNetwork)
-		delete(labels, utils.KeyVlanLabel)
-	}
-
-	if oldConf.Vlan != 0 && oldConf.Vlan != newConf.Vlan {
-		labels[utils.KeyLastVlanLabel] = strconv.Itoa(oldConf.Vlan)
+	err := newConf.SetNetworkInfoToLabels(labels)
+	if err != nil {
+		return nil, err
 	}
 
-	cnName := newConf.BrName[:len(newConf.BrName)-len(iface.BridgeSuffix)]
-	labels[utils.KeyClusterNetworkLabel] = cnName
-	if oldConf.BrName != newConf.BrName {
-		labels[utils.KeyLastClusterNetworkLabel] = oldConf.BrName[:len(oldConf.BrName)-len(iface.BridgeSuffix)]
+	// check and get the bridge name
+	cnName, err := utils.GetBridgeNamePrefix(newConf.BrName)
```

Comment:
nit: we already set cn name to a label during SetNetworkInfoToLabels.

```
lbs[KeyClusterNetworkLabel] = cnname
```

We can just look this from the map and simplify logic